### PR TITLE
Add support for NX-OS fan status

### PR DESCRIPTION
**Pull request by geordish**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

### CODE REVIEW

File: includes/discovery/sensors/states/nxos.inc.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    $fan_trays = snmpwalk_cache_oid_num($device, $fan_tray_oid, array());
+
+    /* CISCO-ENTITY-FRU-CONTROL-MIB cefcFanTrayOperStatus
+     *  unknown(1),
+     *  up(2),
+     *  down(3),
+     *  warning(4)
+     */
+
+    if (is_array($fan_trays)) {
```
Comment (murrant) [inline includes/discovery/sensors/states/nxos.inc.php]: Do you need is_array here? foreach on an empty array does nothing.

Comment (murrant) [inline includes/discovery/sensors/states/nxos.inc.php]: Do you need the is_array() here, foreach on an empty array does nothing.

**Comment by geordish** [comment]
I initially didn't have it. Scrutinizer complained about it, so I added it.

**Comment by laf** [comment]
Will test tomorrow.

File: includes/discovery/sensors/states/nxos.inc.php (hunk 1 above)
Comment (murrant) [inline includes/discovery/sensors/states/nxos.inc.php]: Agreed

**Comment by laf** [comment]
Tested fine but I've moved the additional snmp request so it wasn't run unnecessarily.

**Pull request merged**